### PR TITLE
Add SARIF documentation

### PR DESCRIPTION
**Pull request by paulomorgado**
Add SARIF documentation for tool processing.

### CODE REVIEW

**Comment by paulomorgado** [comment]
I'm missing something because I'm not seeing the generation of the SARIF files.

**Comment by mavasani** [comment]
Did you try running cibuild locally?

File: eng/GenerateAnalyzerNuspec.targets [hunk 1]
```diff
@@ -40,8 +40,13 @@
     </ItemGroup>
 
     <PropertyGroup Condition="'$(GenerateAnalyzerMdFile)' == 'true'">
-      <AnalyzerDocumentationFileDir Condition="'$(AnalyzerDocumentationFileDir)' == ''">$(RepoRoot)src\$(NuspecPackageId)</AnalyzerDocumentationFileDir>
```
Comment (mavasani) [inline eng/GenerateAnalyzerNuspec.targets]: Did you want to revert these deletions? We still want to generate documentation files.

**Comment by paulomorgado** [comment]
I think I got it fixed.

Is there a way to unit test this?

File: src/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+
```
Comment (mavasani) [inline src/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.md]: Why is this file being added? We already have a copy here: https://github.com/dotnet/roslyn-analyzers/blob/master/src/PerformanceSensitiveAnalyzers/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.md

File: src/Microsoft.CodeAnalysis.PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (mavasani) [inline src/Microsoft.CodeAnalysis.PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md]: Same here - this md file already exists, please revert

File: src/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.md (hunk 2 above)
Comment (mavasani) [reply src/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.md]: Also tagging @genlu

**Comment by mavasani** [comment]
> Is there a way to unit test this?

Nope, we just rely on executing the cibuild and validating the generated files are correctly added/updated.

**Comment by paulomorgado** [comment]
I'm already using the SARIF files on my own tooling and they seem fine.

**Comment by mavasani** [comment]
> I'm already using the SARIF files on my own tooling and they seem fine.

My comment was not about the SARIF files added here, but the analyzer documentation files (md files) are getting duplicated in the repo for 2 projects: public API analyzer and performance sensitive analyzers. Basically your PR should only add new sarif files not new .md files.

**Comment by paulomorgado** [comment]
My reply was to the sense that I had validated the contents of the SARIF files.

**Comment by mavasani** [comment]
Ah thanks - can we revert the addition of new .md files in this PR? I can sign off once you do so.

**Comment by paulomorgado** [comment]
I'm sorry, but I'm not getting it.

Which .md files to you want me to remove? Do you want me to remove the corresponding .sarif files?

These were the added and changed .md and .sarif files:

```
A       src/MetaCompilation.Analyzers/MetaCompilation.Analyzers.sarif
A       src/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.sarif
M       src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Microsoft.CodeAnalysis.BannedApiAnalyzers.md
A       src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Microsoft.CodeAnalysis.BannedApiAnalyzers.sarif
M       src/Microsoft.CodeAnalysis.FxCopAnalyzers/Microsoft.CodeAnalysis.FxCopAnalyzers.md
A       src/Microsoft.CodeAnalysis.FxCopAnalyzers/Microsoft.CodeAnalysis.FxCopAnalyzers.sarif
A       src/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.md
A       src/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.sarif
A       src/Microsoft.CodeAnalysis.PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md
A       src/Microsoft.CodeAnalysis.PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.sarif
A       src/Microsoft.CodeAnalysis.VersionCheckAnalyzer/Microsoft.CodeAnalysis.VersionCheckAnalyzer.sarif
A       src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.sarif
M       src/Microsoft.NetCore.Analyzers/Microsoft.NetCore.Analyzers.md
A       src/Microsoft.NetCore.Analyzers/Microsoft.NetCore.Analyzers.sarif
A       src/Microsoft.NetFramework.Analyzers/Microsoft.NetFramework.Analyzers.sarif
M       src/Roslyn.Diagnostics.Analyzers/Roslyn.Diagnostics.Analyzers.md
A       src/Roslyn.Diagnostics.Analyzers/Roslyn.Diagnostics.Analyzers.sarif
```

Looking at it, I assume you want me to remove these:

```
src/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.md
src/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.sarif
src/Microsoft.CodeAnalysis.PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md
src/Microsoft.CodeAnalysis.PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.sarif
```

**Comment by mavasani** [comment]
I sync'ed offline with @paulomorgado and seems the scripting has regressed prior to this PR and we are generating duplicating md files for PublicApiAnalyzers and PerformanceSensitiveAnalyzers. I am fine to sign off on this PR and either me or @genlu will fix this in a follow-up PR.

**Pull request merged**